src/mon/PGMap.h

```diff
@@ -52,7 +52,7 @@ class PGMapDigest {
   mempool::pgmap::map<int64_t,int64_t> num_pg_by_pool;
   pool_stat_t pg_sum;
   osd_stat_t osd_sum;
-  mempool::pgmap::unordered_map<int32_t,int32_t> num_pg_by_state;
+  mempool::pgmap::unordered_map<uint64_t,int32_t> num_pg_by_state;
```

Comment:
Need up update PGMapDigest encode and decode too, so that we can decode the old int32_t instances.  we also need to encode the 32-bit variation in case the mgr upgrades before the mon. this means conditional encoding based on a feature bit... use SERVER_MIMIC!